### ISSUE TITLE
App Distribution Task Doesn't Depend on assemble Task Anymore

### ISSUE THREAD

**Issue by eygraber** [firebase/firebase-android-sdk#3501]
### [REQUIRED] Step 2: Describe your environment

  * Firebase Component: App Distribution Plugin
  * Component version: 3.0.0

### [REQUIRED] Step 3: Describe the problem

After updating to 3.0.0 the upload task runs before the apk is built and fails. 

```
03:34:27  > Task :app:createReleaseApkListingFileRedirect
03:34:27  > Task :app:assembleRelease
03:34:27  
03:34:27  > Task :app:appDistributionUploadRelease FAILED
03:34:27  Using APK path specified by the artifactPath parameter in your app's build.gradle: app/build/outputs/apk/release/app-release.apk.
03:34:27  Uploading APK to Firebase App Distribution...
03:34:33  
03:34:33  
03:34:33  FAILURE: Build failed with an exception.
03:34:33  
03:34:33  * What went wrong:
03:34:33  Execution failed for task ':app:appDistributionUploadRelease'.
03:34:33  > Could not find the APK. Make sure you build first by running ./gradlew assemble[Variant], or set the artifactPath parameter to point to your APK
03:34:33  
03:34:33  * Try:
03:34:33  > Run with --stacktrace option to get the stack trace.
```

#### Steps to reproduce:

```
gradle :app:nightly // see code below for nightly task
```

#### Relevant Code:

```
firebaseAppDistribution {
        serviceCredentialsFile = System.getenv("APP_DISTRIBUTION_KEY")
        groups = "me"
        artifactType = "APK"
        artifactPath = "app/build/outputs/apk/release/app-release.apk"
}
```

Nightly task code
```
tasks.register<MyNightlyTask>("nightly") {
  dependsOn(
        "assembleRelease",
        "appDistributionUploadRelease"
    )
}
```

**Comment by argzdev** [comment]
Hi @eygraber, thanks for reporting the issue with details. I'm just linking this to #2978 for reference.

**Comment by tagboola** [comment]
Hi @eygraber sorry you're running into this issue. Do you get the same error if you don't set the `artifactPath` parameter?

**Comment by eygraber** [comment]
If I don't set `artifactPath` I get the following error:

```
Missing app id. Please check that it was passed in and try again
```

**Comment by tagboola** [comment]
Thanks for the quick response. The `Missing app id. Please check that it was passed in and try again` error is known issue that we plan to fix soon. Can you try to explicitly add `appId` to the `firebaseAppDistribution` block in your `build.gradle` file and see if your upload succeeds? If it does, can you also verify where the output APK file is located?

**Comment by eygraber** [comment]
What value should I use for `appId`?

The output APK file is where `artifactPath` was pointing to. After removing `artifactPath` this was logged:

```
Using APK path in the outputs directory: /home/ubuntu/workspace/nightly/app/build/outputs/apk/release/app-release.apk.
```

**Comment by tagboola** [comment]
I believe the issue is the value you are passing to `artifactPath` is a relative path, when it actually expects an absolute path, which is reflected in the documentation.

![Screen Shot 2022-03-02 at 12 39 14 PM](https://user-images.githubusercontent.com/401051/156417385-07bbe374-25f9-4d92-a1ac-d5b8fba5247a.png)

This apparently wasn't being enforced in previous versions, which goes against the way the parameter is documented. As of `3.0.0` though we are now requiring that it's an absolute path. I don't think this is the best experience though, so we'll look into adding support for relative paths.

As a workaround, you can:
* Downgrade to the previous version
* Remove the `artifactPath` parameter and instead rely on the plugin to infer the path, and also set the  value of `appId` to  your Firebase app ID.

**Comment by eygraber** [comment]
Interesting info about `artifactPath`. I'm probably better off leaving it out then since we're fine with the plugin inferring the path.

Re: `appId`, does that correspond to `mobilesdk_app_id` in my google-services.json?

**Comment by tagboola** [comment]
Yes , it does correspond to the `mobilesdk_app_id` in your `google-services.json`. There is an issue with version `3.0.0` of the plugin where we aren't inferring the `mobilesdk_app_id` correctly, which is why you would have to specify it in your `build.gradle` file. We plan to ship a fix for this soon, though!

**Comment by meladRaouf** [comment]
For me the issue was that I was using relative path for the artifactPath param.
When I used the absloute path the upload works fine

**Comment by rebehe** [comment]
Hi folks, just wanted to let you know that this has been fixed in version `3.0.1` of the appdistribution gradle plugin - upgrade for the fix!

**Comment by argzdev** [comment]
Hi all, since version '3.0.1' resolves this issue, I'll close this now. Thanks!
